pkg/asset/manifests/gcp/cluster.go

```diff
@@ -0,0 +1,150 @@
+	manifests := []*asset.RuntimeFile{}
+
+	const (
+		description = "Created By OpenShift Installer"
+	)
+
+	networkName := fmt.Sprintf("%s-network", clusterID.InfraID)
+	if installConfig.Config.GCP.Network != "" {
+		networkName = installConfig.Config.GCP.Network
+	}
+
+	masterSubnet := gcp.DefaultSubnetName(clusterID.InfraID, "master")
+	if installConfig.Config.GCP.ControlPlaneSubnet != "" {
+		masterSubnet = installConfig.Config.GCP.ControlPlaneSubnet
+	}
+
+	master := capg.SubnetSpec{
+		Name:        masterSubnet,
+		CidrBlock:   "",
+		Description: ptr.To(description),
+		Region:      installConfig.Config.GCP.Region,
+	}
+
+	workerSubnet := gcp.DefaultSubnetName(clusterID.InfraID, "worker")
+	if installConfig.Config.GCP.ComputeSubnet != "" {
+		workerSubnet = installConfig.Config.GCP.ComputeSubnet
+	}
+
+	worker := capg.SubnetSpec{
+		Name:        workerSubnet,
+		CidrBlock:   "",
+		Description: ptr.To(description),
+		Region:      installConfig.Config.GCP.Region,
+	}
+
+	// Add the CIDR information.
+	machineV4CIDRs := []string{}
+	for _, network := range installConfig.Config.Networking.MachineNetwork {
+		if network.CIDR.IPNet.IP.To4() != nil {
+			machineV4CIDRs = append(machineV4CIDRs, network.CIDR.IPNet.String())
+		}
+	}
+
+	if len(machineV4CIDRs) == 0 {
+		return nil, fmt.Errorf("failed to parse machine CIDRs")
+	}
+
+	_, ipv4Net, err := net.ParseCIDR(machineV4CIDRs[0])
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse machine network CIDR: %w", err)
+	}
+
+	if installConfig.Config.GCP.ControlPlaneSubnet == "" {
+		masterCIDR, err := cidr.Subnet(ipv4Net, 1, 0)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create the master subnet %w", err)
+		}
+		master.CidrBlock = masterCIDR.String()
+	}
+
+	if installConfig.Config.GCP.ComputeSubnet == "" {
+		workerCIDR, err := cidr.Subnet(ipv4Net, 1, 1)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create the worker subnet %w", err)
+		}
+		worker.CidrBlock = workerCIDR.String()
+	}
+
+	subnets := []capg.SubnetSpec{master, worker}
+
+	labels := map[string]string{}
+	labels[fmt.Sprintf("kubernetes-io-cluster-%s", clusterID.InfraID)] = "owned"
+	labels[fmt.Sprintf("capg-cluster-%s", clusterID.InfraID)] = "owned"
+	for _, label := range installConfig.Config.GCP.UserLabels {
+		labels[label.Key] = label.Value
+	}
+
+	// Find the availability zones and use this information to configure the GCP Cluster Failure Domains.
+	zones := sets.New[string]()
+	if installConfig.Config.Platform.GCP != nil {
+		if installConfig.Config.Platform.GCP.DefaultMachinePlatform != nil {
+			for _, zone := range installConfig.Config.Platform.GCP.DefaultMachinePlatform.Zones {
+				zones.Insert(zone)
+			}
+		}
+	}
+	if len(zones) == 0 {
+		// Find the possible zones from the Control Plane
+		if installConfig.Config.ControlPlane.Platform.GCP != nil {
+			for _, zone := range installConfig.Config.ControlPlane.Platform.GCP.Zones {
+				zones.Insert(zone)
+			}
+		}
```

Comment:
I think it makes more sense to set failure domains only when `installConfig.Config.ControlPlane.Platform.GCP` and ` installConfig.Config.Compute[0].Platform.GCP` are set

Reply:
Do we still want to set any failure domains from the `DefaultMachinePlatform.Zones`? If so, do we only use these zones when the above two are not set?

Reply:
> Do we still want to set any failure domains from the DefaultMachinePlatform.Zones?

Yes,  always use `DefaultMachinePlatform`, it is the equivalent of setting both. If it is set, we should still add in any zones that are set for either pool.